site/docs/expressions/lambda_expressions.md

```diff
@@ -0,0 +1,137 @@
+# Lambda Expressions
+
+Lambda expressions represent inline, anonymous functions within query plans, enabling operations that require nested computation over data. They are primarily used to express array operations (such as `transform` or `filter`) and support higher-order functions that apply computations element-wise over collections. A lambda consists of explicit parameter types and a body expression that references those parameters.
+
+## Overview
+
+Lambda expressions are a type of expression in Substrait (like `IfThen`, `Subquery`, or `Nested` expressions) that can be passed as arguments to higher-order functions or [invoked directly](#lambda-invocation).
+
+!!! note "Documentation Syntax"
+    This documentation uses the syntax `(param: type, ...) -> expression` as an illustrative notation to explain lambda concepts in a readable form. There is no formal syntax specified in the Substrait spec for compactly representing lambdas.
```

Comment:
This is kind of true of everything though right?

Also, we do have formal syntax in some contexts (e.g. extensions files)

Reply:
I'll defer to your preference.  It doesn't really add or subtract IMO.